junit/src/main/java/org/kaazing/k3po/junit/rules/SpecificationStatement.java

```diff
@@ -68,39 +68,43 @@ public void evaluate() throws Throwable {
             try {
                 // note: JUnit timeout will trigger an exception
                 statement.evaluate();
-            } catch (AssumptionViolatedException e) {
-
+            } catch (Throwable cause) {
+                // any exception aborts the script (including timeout)
                 if (!latch.isFinished()) {
                     scriptRunner.abort();
                 }
 
-                throw e;
-            } catch (Throwable cause) {
-                // any exception aborts the script (including timeout)
+                // wait at most 5sec for the observed script (due to the abort case)
+                // should take less than a second for K3PO to complete
+                ScriptPair scripts = null;
```

Comment:
I think this logic is correct, but the code complexity is high (mostly due to legacy problems).  If possible I would like to try to simplify it.